lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -783,16 +870,20 @@ def add_path(self, path: str, *, recursive: bool = False):
             path: either a rules file or a dir of rules files.
             recursive: whether to read files recursively or not (no impact if `path` is a file).
 
-        Raises:
-            InvalidAlertRulePathError: if the provided path is invalid.
+        Returns:
+            True if path was added else False.
         """
-        path = Path(path)  # type: Path
-        if path.is_dir():
-            self.alert_groups.extend(self._from_dir(path, recursive))
-        elif path.is_file():
-            self.alert_groups.extend(self._from_file(path.parent, path))
-        else:
-            raise InvalidAlertRulePathError(path, "path does not exist")
+        rules_path = Path(path)
+        if not rules_path.exists:
+            logger.warning("Alert rules path %s does not exist", path)
+            return False
+
+        if rules_path.is_dir():
+            self.alert_groups.extend(self._from_dir(rules_path, recursive))
+        elif rules_path.is_file():
+            self.alert_groups.extend(self._from_file(rules_path.parent, rules_path))
+
+        return True
```

Comment:
The real issue here is the fact that `add_path()` was designed to "return" success/failure information but the methods it depends on `_from_file()` and `_from_dir()` were not. Hence `add_path()` does not have enough information to determine if the path was "truly added". My two cents -- let us ignore this for beta since this is an internal implementation detail and pick it up later.